src/components/organisms/Top.tsx

```diff
@@ -75,20 +81,26 @@ export default function Top({ deviceId }: Props) {
             humidity={weatherData.humidity}
             weather_icon={weatherData.weather_icon}
           />
-        ) : (
+        ) : weatherErrMessage ? (
           <div className={styles.top__weather}>
             Failed to retrieve the weather data.
           </div>
+        ) : (
+          <div className={styles.top__weather}>Loading the weather data.</div>
         )}
         {liveData ? (
           <Livedata
             temperature_celsius={liveData.temperature_celsius}
             humidity={liveData.humidity}
           />
-        ) : (
+        ) : liveErrMessage ? (
           <div className={styles.top__livedata}>
             Failed to retrieve the temperature and humidity data.
           </div>
```

Comment:
I think you should replace the text `Failed to retrieve the temperature and humidity data.` to {liveErrMessage}